polaris/experimental/_dataset_v2.py

```diff
@@ -161,6 +147,31 @@ def has_zarr_manifest_md5sum(self) -> bool:
         """Whether the md5sum for this dataset's zarr manifest has been computed and stored."""
         return self._zarr_manifest_md5sum is not None
 
+    @computed_field
+    @property
+    def md5sum(self) -> str:
+        """
+        Lazily compute the checksum once needed.
+
+        The checksum of the DatasetV2 is computed from the Zarr Manifest and is _not_ deterministic.
+        """
+        if not self.has_md5sum:
+            logger.info("Computing the checksum. This can be slow for large datasets.")
+            self.md5sum = calculate_file_md5(self.zarr_manifest_path)
+        return self._md5sum
+
+    @md5sum.setter
+    def md5sum(self, value: str):
+        """Set the checksum."""
+        if not re.fullmatch(r"^[a-f0-9]{32}$", value):
+            raise ValueError("The checksum should be the 32-character hexdigest of a 128 bit MD5 hash.")
+        self._md5sum = value
+
+    @property
+    def has_md5sum(self) -> bool:
+        """Whether the md5sum for this class has been computed and stored."""
+        return self._md5sum is not None
+
```

Comment:
**Comment:** To clearly mark the V2 implementation as a departure from the V1 implementation, we renamed the `_md5sum` property for V2 to `_zarr_manifest_md5sum`. These methods and the `_md5sum` property are thus not needed.